src/trace-plugin-interface.js

```diff
@@ -0,0 +1,246 @@
+'use strict';
+var TraceLabels = require('./trace-labels.js');
+var cls = require('./cls.js');
+var constants = require('./constants.js');
+var is = require('is');
+
+/**
+ * This file describes an interface for third-party PluginAPIs to enable tracing
+ * for arbitrary modules.
+ */
+
+/**
+ * An object that is associated with a single child span. It exposes
+ * functions for adding labels to or closing the associated span.
+ */
+function ChildSpan(agent, context) {
+  this.agent = agent;
+  this.context = context;
```

Comment:
Yes, it's a misnomer here.

I'll also make these private, ideally plugins should not access them other than in exceptional circumstances.